exercises/react/source/react.d

```diff
@@ -175,12 +175,12 @@ static if (allTestsEnabled) {
     auto b = r.new InputCell(false);
     auto carryIn = r.new InputCell(false);
 
-    auto aXorB = r.new ComputeCell(a, b, (x, y) => x != y);
-    auto sum = r.new ComputeCell(aXorB, carryIn, (x, y) => x != y);
+    auto aXorB = r.new ComputeCell(a, b, (int x, int y) => x != y);
```

Comment:
since this is a `Reactor!(bool)`, implying that the value of each cell is of type `bool`, how about having this be `(bool x, bool y)` instead of `(int x, int y)`? I understand that it is possible to implicitly convert between `int` and `bool`, but given that we know that the types are `bool`, it seems like we can be specific here and not use the implicit conversion.

(Same comment applies to all the functions listed in this test case; they could all be `bool` instead of `int`)

Reply:
My bad, should be int.

Reply:
D function literals are templates unless passed to a function expecting a specific type of function pointer.

The solutions to this are as follows:
### Rearrange all constructor signatures to take the function first 
This enables enables the following pattern
```D
class pack {
    this(G...)(G list)
    {
       
    }
}
class test {
	this(T...)(int function(T) y, pack oof ...)
    {
    //Type checking by hand unfortunately 
    }
}
void main()
{
    auto y = new test( (int x) => x + 1, 23, 45);
}
```
I think this works, but I haven't tried it properly. By "works", I mean can be resolved solely at compile time.

### Alias Templates
All (to my knowledge &  where possible) parts of the D standard library use templates that take function parameters by alias (or as a variadic template pack of unit length). 
```D
class pack(alias T) {
	void exec()
    {
        static if(__traits(compiles, T())) {
        	T();
        }
    }
}
void func()
{

}
void main()
{
    auto y = new pack!((int x) => x + 1);
    auto g = new pack!func;
}
```

I'm not sure an alternative, complete,  solution really exists. - Other than providing the parameters explicitly - because it's not the way that particular feature of D is designed to work. 

Note: By changing the API (return a voldemort struct with, say, opCall defined taking the function [Or the other way around] - this is effectively Currying but done manually ), this could probably be done safely and generically.

Reply:
I checked and the tests still do compile with the example solution when using `(bool x, bool y)` so I would indeed suggest amending this PR to use that (unless I missed some reason why it should stay `int`)

Reply:
I'm going to but I did the edit on GitHub directly so i need to find the file (my fork hasn't updated which odd).

Reply:
If it helps, I see the file on your fork at https://github.com/maxhaton/d/blob/patch-1/exercises/react/source/react.d

Reply:
Cheers. Didn't check the branches

Reply:
@petertseng  fixed